### PR TITLE
chore(ci): Add comment with summary for changed tables

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/3557.
First step for https://github.com/cloudquery/cloudquery/issues/2769.

Example in https://github.com/cloudquery/cloudquery/pull/5375. I chose a flat list, sorted by tables, then for each table breaking changes come first.
I think a flat list can make it easier to add changelog entries to the PR, but we can always update the format of the comment.

I'll follow up next with a PR to fail renovate updates if there's a breaking change in a non major version (https://github.com/cloudquery/cloudquery/issues/2769).

Another possible solution is to use the plugins `doc --format json` command, but that requires checking out the code from `main`, running the command for each plugin, then doing the same for the PR branch, then comparing the JSON files.

> Possible improvement, detect order changes to PKs as breaking

### CODE REVIEW

File: .github/workflows/docs_changes_summary.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+jobs:
+  doc-changes:
+    defaults:
+      run:
+        working-directory: scripts/table_diff
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (hermanschaaf) [inline .github/workflows/docs_changes_summary.yml:18]: Naive question: does this not check out the PR code when used on `pull_request_target`?

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
Looks great! Just want to check I'm not missing something about the safety of pull_request_target when it comes to checking out code

File: .github/workflows/docs_changes_summary.yml (hunk 1 above)
Comment (erezrokah) [reply .github/workflows/docs_changes_summary.yml:18]: `pull_request_target` runes in the context of the `main` branch so this checks out the `main` branch code.
In order to checkout the PR's code, one needs to pass the PR `sha` explicitly by doing:
```
- uses: actions/checkout@v2
      with:
        ref: ${{ github.event.pull_request.head.sha }}
```

more in https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

**Comment by erezrokah** [comment]
> Just want to check I'm not missing something about the safety of pull_request_target when it comes to checking out code

Thanks for the review, and great point. Commented on the `pull_request_target` usage

File: .github/workflows/docs_changes_summary.yml (hunk 1 above)
Comment (hermanschaaf) [reply .github/workflows/docs_changes_summary.yml:18]: Ah okay, got it, thanks for the explanation!

**Pull request merged**